### PR TITLE
Fix bad numbering format (.1.) for section numbers (sectnums) and some...

### PR DESCRIPTION
**Pull request by michelle-purcell**
This PR fixes the problem of an extra dot (.) before each section number, which appears in tutorials with `:sectnums:` enabled.

image

and 

image

This update:

- Adds the `:numbered:` AsciiDoc attribute to the doc header of a tutorial and the tutorial template file to enforce the Arabic number format of 1. 2. 3. etc etc instead of .1.  .2. .3.  etc, which sometimes occurs depending on the markup.
- Fixes the two OIDC security tutorials, which have bad numbering rendered on the doc site.

image
 - Also fixes a code snippet rendering issue spotted by @gsmet during this review.

### CODE REVIEW

**Comment by michelle-purcell** [comment]
@sheilamjones - Thanks so much for flagging this problem :+1: 
 The fix in this PR looks good both up and downstream from what I have tested locally.

**Comment by michelle-purcell** [comment]
@maxandersen @aloubyansky - When this update is merged to `main`, please backport to 3.2 for product docs. It's a doc bug fix for both upstream and downstream doc rendering. Thanks so much :pray:

**Comment by gsmet** [comment]
From what I can see, this is not the only problem with this guide on the website.

The tabs are broken in the `Using Maven (pom.xml)` and `Using Gradle (build.gradle)` sections.

My understanding was that you were supposed to use the common pattern we had everywhere else and have a transformation to make it compatible with downstream but I see that you are using a completely different pattern that is breaking the Javascript.

**Comment by gsmet** [comment]
See 
2023-11-07 22 07 00  7b5fd49843c1

**Comment by sheilamjones** [comment]
@gsmet, thanks for your review. This PR is to address the numbering issue only. 
I am processing a separate PR to address final edits to this Bearer token guide (https://github.com/quarkusio/quarkus/pull/35849/files). I noticed the issue with this code snippet also and for the purposes of having an atomic commit, shall address that code snippet issue there. I hope this okay.

**Comment by michelle-purcell** [comment]
@gsmet - Thanks for reviewing and the above ^^ @gsmet. 
+1 to comment ^^ from @sheilamjones. Can we handle the other issue in a different PR? Thanks.

**Comment by maxandersen** [comment]
> See

which guide is that from ? didn't spot such rendering on the preview or in quarkus.io?

File: docs/src/main/asciidoc/_templates/template-tutorial.adoc [hunk 1]
```diff
@@ -12,6 +12,7 @@ TODO:
 [id="..."]
 = Title using sentence capitalization
 include::_attributes.adoc[]
+:numbered:
```
Comment (maxandersen) [inline docs/src/main/asciidoc/_templates/template-tutorial.adoc:15]: is this issue of :numbered: not general? 

should it be set in pom.xml attributes section instead?

**Comment by michelle-purcell** [comment]
@maxandersen 

> > See
> 
> which guide is that from ? didn't spot such rendering on the preview or in quarkus.io?


Me neither.

The numbering issue, which this PR aims to fix, appears in this tutorial:

https://quarkus.io/guides/security-oidc-bearer-token-authentication-tutorial

For consistency, I applied the same approach to this tutorial also:

https://quarkus.io/guides/security-oidc-code-flow-authentication-tutorial

Can we use this PR to focus on the numbering issue? I will open a new PR for the other code issue. Thanks.

**Comment by michelle-purcell** [comment]
@gsmet / @maxandersen / @aloubyansky - In the interest of closing this and getting the topic to QE ASAP for 3.2.8 product doc release, I fixed the code snippet issue that @gsmet raised during this review in this PR. Can we merge and backport now?
Thanks all.

**Comment by sheilamjones** [comment]
Hi @gsmet / @maxandersen / @aloubyansky, would it be possible to merge and backport this PR please as a separate PR with some additional Vale fixes for this same topic is dependent on it?
Many thanks,
Sheila

**Comment by aloubyansky** [comment]
I am planning another backport round tomorrow. But I'd like somebody more familiar with the doc formatting review this one. Thanks.

**Comment by michelle-purcell** [comment]
@aloubyansky / @gastaldi / @gsmet gsmet - Please approve and merge the content fixes. We no longer need this to be backported but it would be good to make the fix in main as the doc defect exists here and needs to be fixed also. Thanks.

**Review by gastaldi — APPROVED** [review]

File: docs/src/main/asciidoc/_templates/template-tutorial.adoc (hunk 1 above)
Comment (gastaldi) [reply docs/src/main/asciidoc/_templates/template-tutorial.adoc:15]: Ah, that's a valid concern, any thoughts @michelle-purcell ?

**Comment by sheilamjones** [comment]
Hi @aloubyansky / @gastaldi / @gsmet, would it be possible to please approve and merge these fixes in Main? Note that we no longer need to backport this, but it would be good to make these fixes in main.
Many thanks.

File: docs/src/main/asciidoc/_templates/template-tutorial.adoc (hunk 1 above)
Comment (aloubyansky) [reply docs/src/main/asciidoc/_templates/template-tutorial.adoc:15]: @michelle-purcell could you please respond to the above question?

Comment (michelle-purcell) [reply docs/src/main/asciidoc/_templates/template-tutorial.adoc]: @maxandersen - Sorry, I missed this Q... 

Yes, I think you are right @maxandersen - Adding the header attribute `numbered` fixed the .1. numbering part of the issue but wasn't enough to fix the indentation level 3-4 part of the numbering issue. The latest edits also resolve the numbering format issue too. I just tested without that attribute and the numbering works fine locally so I removed it from the 2 modified topics and the tutorial template. Thanks :star:

**Comment by michelle-purcell** [comment]
@maxandersen - All comments are now resolved. Can we merge this one to `main`? Thanks.

**Pull request merged**